### PR TITLE
feat: add redux logic for projects page, including loading existing p…

### PR DESCRIPTION
**Pull request by hazigabriel**
…rojects and adding new projects

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -23,10 +25,11 @@ function App() {
         <AuthProvider>
           <Routes>
             <Route path="/" element={<LayoutPage />}>
-              <Route index element={<Projects />} />
+              <Route index element={<Navigate to ='projects'/>} />
```
Comment (rexalex) [inline src/App.js]: `<Navigate to="projects" />`
please change :)

File: src/App.js [hunk 2]
```diff
@@ -23,10 +25,11 @@ function App() {
...
               <Route path="/projects" element={<Projects />} />
+              <Route path="/projects/add-project" element={<NewProject />} />
               <Route path="/projects/:projectId" element={<ProjectItem />} />
-              <Route
+               <Route
```
Comment (rexalex) [inline src/App.js]: please delete extra space

File: src/app/store.js [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
+import projectsReducer from "../features/projects/projectsSlice"
 
 export const store = configureStore({
 	reducer: {
-		auth: authReducer
-	}
+		auth: authReducer,
+		projects: projectsReducer
+ 	}
 })
```
Comment (rexalex) [inline src/app/store.js]: please use indent using spaces 2

File: src/components/LayoutPage/LayoutPage.js [hunk 4]
```diff
@@ -90,11 +90,8 @@ export default function LayoutPage() {
...
-
-						<Col span={4}>
-							{/* <Siderbar /> */}
-						</Col>
-						<Col span={20}>
+ 
+						<Col span={24}>
 							<Outlet />
 							{/* reference for outlet > https://www.youtube.com/watch?v=PWi9V9d_Jsc */}
 						</Col>
```
Comment (rexalex) [inline src/components/LayoutPage/LayoutPage.js:97]: please use indent using spaces 2

File: src/components/LayoutPage/pages/Projects/NewProject/NewProject.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+						</Form.Item>
+						<Form.Item data-cy="newProjectSubmitButton" >
+							<Button type="primary" htmlType="submit">Create project</Button>
+						</Form.Item>
+					</Form>
+				</Card>
+			</Layout.Content>
+		</Layout>
+	)
+}
```
Comment (rexalex) [inline src/components/LayoutPage/pages/Projects/NewProject/NewProject.js:181]: please use indent using spaces 2

File: src/components/LayoutPage/pages/Projects/Projects.js [hunk 6]
```diff
@@ -1,7 +1,104 @@
...
+			</Row>
+			<Row gutter={[16, 16]} >
+				{projectList ? projectList.map((project, iterationId) => {
+					{ return generateProjectCard(project, iterationId) }
+				}) : 'No projects added yet'}
+			</Row>
+			<Outlet />
+		</Layout>
 	)
 }
```
Comment (rexalex) [inline src/components/LayoutPage/pages/Projects/Projects.js:104]: please use indent using spaces 2

File: src/components/LayoutPage/pages/UserList/UserList.js [hunk 7]
```diff
@@ -84,13 +84,9 @@ export default function UserList() {
...
-
-
-
 					{userList.map((user, id) => {
 						return userGenerator(user.name, user.phoneNumber, user.email, id)
 					})}
-
 				</Collapse>
 			</Layout>
 		</Layout>
```
Comment (rexalex) [inline src/components/LayoutPage/pages/UserList/UserList.js:92]: please use indent using spaces 2

File: src/features/projects/projectsService.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+	return response.data
+}
+
+const projectsService = {
+	getProjects,
+	addProject
+}
+
+export default projectsService
```
Comment (rexalex) [inline src/features/projects/projectsService.js]: please use indent using spaces 2

File: src/features/projects/projectsService.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import axios from "axios";
+const ALL_PROJECTS_URL = "http://localhost:8080/api/projects/"
+
+//get all projects
+const getProjects = async () => {
+	const PROJECTS_URL = "http://localhost:8080/api/projects/"
```
Comment (rexalex) [inline src/features/projects/projectsService.js]: is there any difference between PROJECTS_URL and ALL_PROJECTS_URL
ALL_PROJECTS_URL is not used
create all constants fro url outside

File: src/features/projects/projectsSlice.js [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+				state.isError = true
+				state.message = action.payload
+			})
+ 
+	},
+})
+
+export const projectsActions = projectsSlice.actions;
+export const { reset } = projectsSlice.actions
+export default projectsSlice.reducer
```
Comment (rexalex) [inline src/features/projects/projectsSlice.js]: please use indent using spaces 2

**Review by rexalex — CHANGES_REQUESTED** [review]
please apply changes left in comments :)

File: src/features/projects/projectsService.js (hunk 9 above)
Comment (hazigabriel) [reply src/features/projects/projectsService.js]: ALL_PROJECTS_URL was left here in error, I've made the changes requested

**Pull request merged**